### PR TITLE
Maintain FlutterTextInputView text in NSMutableString

### PR DESCRIPTION
**Pull request by cbracken**
We were converting from/to NSString at the interface of
FlutterTextInputView; instead use NSMutableString to maintain the
internal buffer.

### CODE REVIEW

**Comment by cbracken** [comment]
Fixes flutter/flutter#9146

**Comment by cbracken** [comment]
Good catch! Done.

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 1]
```diff
@@ -111,18 +113,11 @@ - (void)deleteBackward {
...
     start -= 1;
     len = 1;
-    if (start > 0 && UTF16_IS_LEAD(_text[start - 1]) &&
-        UTF16_IS_TRAIL(_text[start])) {
-      start -= 1;
-      len += 1;
-    }
-    _text.erase(start, len);
   }
+  [self.text deleteCharactersInRange:NSMakeRange(start, len)];
```
Comment (abarth) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Does this handle surrogates properly?

Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Agh; after a local test -- no. I'll put together a patch.

Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Retro-comment for archaeologists: patched in #3557.

**Pull request merged**